examples/arithmetics/src/language-server/main-browser.ts

```diff
@@ -26,14 +28,20 @@ const documentChangeNotification = new NotificationType<DocumentChange>('browser
 const jsonSerializer = arithmetics.serializer.JsonSerializer;
 shared.workspace.DocumentBuilder.onBuildPhase(DocumentState.Validated, documents => {
     for (const document of documents) {
-        const json = jsonSerializer.serialize(document.parseResult.value, {
-            sourceText: true,
-            textRegions: true,
-        });
+        const json = [];
+        const module = document.parseResult.value as Module;
+        if(document.diagnostics === null || document.diagnostics!.filter((i) => i.severity === 1).length === 0) {
+            for (const [evaluation, value] of interpretEvaluations(module)) {
+                const cstNode = evaluation.expression.$cstNode;
+                if (cstNode) {
+                    json.push({range: evaluation.expression.$cstNode?.range, text: evaluation.expression.$cstNode?.text, value: value});
+                }
+            }
+        }
         connection.sendNotification(documentChangeNotification, {
             uri: document.uri.toString(),
-            content: json,
+            content: JSON.stringify({ ast: jsonSerializer.serialize(module), evaluations: json }),
```

Comment:
That's a tricky change, as it affects my envisioned code generation demo (which you can't be aware of 😉).
Please keep the AST serialization as it was above including the instruction of
```javascript

{
    sourceText: true,
    textRegions: true,
}
```
Question: Could do the interpretation also on client side?
With the serialized AST including the original source text and the text regions of the particular AST elements that shouldn't be difficult. Let's chat about this if you like.

Reply:
Thanks, I added the JSON part back! 
I actually talked about this and we concluded that sending the results would be more efficient for the preview. 
Also for people looking at the example, it's helpful to know how to process the last inside of the language server